dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -455,37 +475,73 @@ public void close() {}
 
     private static final class ResponseInputStream extends FilterInputStream {
 
-        // Client reference is used to prevent premature termination
-        @Nullable
-        private ApacheHttpClientChannels.CloseableClient client;
-
         private final HttpClientResponse response;
 
-        ResponseInputStream(
-                @Nullable ApacheHttpClientChannels.CloseableClient client,
-                InputStream stream,
-                HttpClientResponse response) {
+        ResponseInputStream(InputStream stream, HttpClientResponse response) {
             super(stream);
-            this.client = client;
             this.response = response;
         }
 
         @Override
-        public void close() throws IOException {
-            try {
-                try {
-                    response.close();
-                } finally {
-                    super.close();
-                }
-            } finally {
-                client = null;
+        public int read() throws IOException {
+            checkOpen();
+            return super.read();
+        }
+
+        @Override
+        public int read(byte[] buffer) throws IOException {
+            checkOpen();
```

Comment:
How expensive are these checkOpen calls?

Reply:
Would the delegate not already do these checks for you?

Reply:
It’s just a null check, idea is to co solo safe state between response.close and stream.close which wouldn’t be guaranteed to agree otherwise